### PR TITLE
[GPU] Fix a bug of logical padding of convolution

### LINKED ISSUE #9039: [Bug] Padded convolution - All zeros on GPU
##### System information (version)
- OpenVINO=> 2021.4.752
- Operating System / Platform =>  Windows 64 Bit
- Compiler :grey_question: => from installer + python-3.7.11 - anaconda3 64 Bit
- Problem classification => Model Conversion :grey_question:

##### Detailed description
I have a super simple ONNX model generated from pytorch like below - A 1D convolution with a padding of 1. 
When i import it in OpenVino-python and i try to run inference + compare outputs i get all zeros when run on GPU, the correct result is instead inferred on CPU.

When instead the 

##### Steps to reproduce
Generate ONNX + save expected output for input, like so (or download the 3 files: [files.zip](https://github.com/openvinotoolkit/openvino/files/7660983/files.zip)):

```
import numpy as np
import torch
from torch import nn

module_pt = nn.Conv1d(in_channels=78,
                      out_channels=64,
                      kernel_size=3,
                      padding=1,
                      bias=False)

np.random.seed(1234)
data = np.random.random(size=(1, 78, 150))
data_t = torch.from_numpy(data).to(torch.float32)

ref_out = module_pt(data_t)

np.save("input.npy", data)
np.save("output.npy", ref_out.detach().cpu().numpy())
torch.onnx.export(module_pt, data_t, "model.onnx")
```

Run the following code.

In brief, this code loads the onnx network, loads the numpy array for the input tensor, runs inference and compares with the numpy array loaded as reference output. Repeat for device = "GPU"

```
import numpy as np
from openvino.inference_engine import IECore

ie = IECore()
net = ie.read_network(model="model.onnx")

sample_input = np.load("input.npy")
output_ref = np.load("output.npy")

input_blob = next(iter(net.input_info))
out_blob = next(iter(net.outputs))

for device in ["CPU", "GPU"]:
    exec_net = ie.load_network(network=net, device_name=device)

    output_test = exec_net.infer(inputs={input_blob: sample_input})[out_blob]

    np.testing.assert_allclose(output_test, output_ref, rtol=1e-03, atol=1e-03), (output_test, output_ref)

    print(device, "ok")
```

the test asserts because the output of the convolution on GPU is all zeros:

```
CPU ok

AssertionError:
Not equal to tolerance rtol=0.001, atol=0.001

Mismatched elements: 9579 / 9600 (99.8%)
Max absolute difference: 1.2720033
Max relative difference: 1.
 x: array([[[0., 0., 0., ..., 0., 0., 0.],
        [0., 0., 0., ..., 0., 0., 0.],
        [0., 0., 0., ..., 0., 0., 0.],...
 y: array([[[-0.64862 , -0.382263, -0.505221, ..., -0.626626, -0.548777,
         -0.2749  ],
        [-0.053266, -0.033243, -0.057755, ...,  0.344207, -0.247998,...
```

When there is no dilation the issue doesn't persist!

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues, Stack Overflow, etc and have not found solution
   
 - [x] There is reproducer code and related data files: images, videos, models, etc.

### PR DESCRIPTION
**Pull request by jade-cho**
+ Fix that it transforms logical padding with wrong axis for 3D tensor.
+ Fix a typo issue.

### CODE REVIEW

File: src/plugins/intel_gpu/src/plugin/ops/convolution.cpp [hunk 1]
```diff
@@ -52,9 +52,9 @@ static ConvoltuionParameters GetConvolutionParameters(const ngraph::CoordinateDi
             break;
         }
         case 1: {
-            stride = cldnn::tensor(cldnn::batch(1), cldnn::feature(1), cldnn::spatial(strides[0], 1, 1));
-            padding = cldnn::tensor({0, 0, TensorValue(pads_begin[0]), 0}, 0);
-            dilation = cldnn::tensor(cldnn::batch(1), cldnn::feature(1), cldnn::spatial(dilations[0], 1, 1));
+            stride = cldnn::tensor(cldnn::batch(1), cldnn::feature(1), cldnn::spatial(1, strides[0], 1));
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/plugin/ops/convolution.cpp:55]: I don't really understand why this 1d dimension must be set to y instead of x. Could you explain more?

Comment (jade-cho) [reply src/plugins/intel_gpu/src/plugin/ops/convolution.cpp:55]: Please see this file: src/plugins/intel_gpu/include/intel_gpu/plugin/common_utils.hpp
tensor_from_dims() function makes 3D input tensor (b, f, x) to 4D tensor (b, f, 1, y). So stride, padding and dilation should be changed from 1D to 2D in the same axis order like input tensor.

inline cldnn::tensor tensor_from_dims(const InferenceEngine::SizeVector& dims, int def = 1) {
    switch (dims.size()) {
    case 0: return cldnn::tensor(cldnn::batch(def), cldnn::feature(def), cldnn::spatial(def, def));
    case 1: return cldnn::tensor(cldnn::batch(dims[0]), cldnn::feature(def), cldnn::spatial(def, def));
    case 2: return cldnn::tensor(cldnn::batch(dims[0]), cldnn::feature(dims[1]), cldnn::spatial(def, def));
    case 3: return cldnn::tensor(cldnn::batch(dims[0]), cldnn::feature(dims[1]), cldnn::spatial(def, dims[2]));
    case 4: return cldnn::tensor(cldnn::batch(dims[0]), cldnn::feature(dims[1]), cldnn::spatial(dims[3], dims[2]));
    case 5: return cldnn::tensor(cldnn::batch(dims[0]), cldnn::feature(dims[1]), cldnn::spatial(dims[4], dims[3], dims[2]));
    case 6: return cldnn::tensor(cldnn::batch(dims[0]), cldnn::feature(dims[1]), cldnn::spatial(dims[5], dims[4], dims[3], dims[2]));
    default: IE_THROW() << "Invalid dimensions size(" << dims.size() << ") for gpu tensor";
    }
}

**Review by vladimir-paramuzov — APPROVED** [review]
